registry/job.go

```diff
@@ -127,18 +130,14 @@ func (r *EtcdRegistry) Units() ([]job.Unit, error) {
 		if u == nil {
 			continue
 		}
-		uMap[u.Name] = u
-	}
-
-	var sortable sort.StringSlice
-	for name, _ := range uMap {
-		sortable = append(sortable, name)
-	}
-	sortable.Sort()
+		// Just a guard
+		_, ok := uMap[u.Name]
+		if ok {
+			continue
+		}
```

Comment:
Right, I don't understand - what are we guarding against? in what situation do we get dupes?